src/graph/entities/qg_node.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2018-2019 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Redis Labs Source Available License Agreement
+ */
+
+#include "qg_node.h"
+#include "../graph.h"
+#include "../../util/arr.h"
+#include <assert.h>
+
+QGNode* QGNode_New(const char *label, const char *alias) {
+    QGNode *n = rm_malloc(sizeof(QGNode));
+    n->alias = NULL;
+    n->label = NULL;
+    n->labelID = GRAPH_NO_LABEL;
+    n->incoming_edges = array_new(QGEdge*, 0);
+    n->outgoing_edges = array_new(QGEdge*, 0);
+    // n->id = -1;
+    return n;
+}
+
+void QGNode_ConnectNode(QGNode* src, QGNode* dest, QGEdge* e) {
+	src->outgoing_edges = array_append(src->outgoing_edges, e);
+	dest->incoming_edges = array_append(dest->incoming_edges, e);
+}
+
+void _QGNode_RemoveEdge(QGEdge **edges, QGEdge *e) {
+	uint edge_count = array_len(edges);
+	for(uint i = 0; i < edge_count; i++) {
+		QGEdge *ie = edges[i];
+		if(e == ie) {
+			edges[i] = edges[edge_count-1];
```

Comment:
Consider updating arr.h to get array_del

Reply:
:+1: I looked over the redisearch implementation; it looks great - I'll bring it in here soon.